consensus/service/src/consensus_service.rs

```diff
@@ -636,70 +646,165 @@ impl<
     /// Helper method for creating the get config json function needed by the GRPC admin service.
     fn create_get_config_json_fn(&self) -> GetConfigJsonFn {
         let ledger_db = self.ledger_db.clone();
-        let byzantine_ledger = self
+
+        let byzantine_ledger: Weak<OnceCell<ByzantineLedger>> = self
             .byzantine_ledger
             .as_ref()
             .map(Arc::downgrade)
             .expect("Server was not initialized");
+
         let config = self.config.clone();
         let logger = self.logger.clone();
         Arc::new(move || {
-            let mut sync_status = "synced";
-            let mut peer_block_height: u64 = 0;
-            byzantine_ledger.upgrade().map(|ledger| {
-                ledger.get().map(|ledger| {
-                    if ledger.is_behind() {
-                        sync_status = "catchup";
-                    };
-                    peer_block_height = ledger.highest_peer_block();
-                })
-            });
-            let block_height;
-            let latest_block_hash;
-            let latest_block_timestamp;
-            let blocks_behind;
-            // If we do not get a num_blocks, several status points will be null
-            match ledger_db.num_blocks() {
-                Ok(b) => {
-                    block_height = Some(b);
-                    latest_block_hash = ledger_db
-                        .get_block(b - 1)
-                        .map(|x| hex::encode(x.id.0))
-                        .map_err(|e| log::error!(logger, "Error getting block {} {:?}", b - 1, e))
-                        .ok();
-
-                    latest_block_timestamp = match ledger_db.get_block_signature(b - 1) {
-                        Ok(x) => Some(x.signed_at()),
-                        // Note, a block signature will be missing if the corresponding block was not
+            // The highest block in the local ledger, and its BlockSignature if available.
+            let (highest_block, highest_block_signature): (Block, Option<BlockSignature>) = {
+                // Number of blocks in the local ledger.
+                let num_blocks: u64 = ledger_db.num_blocks().map_err(|e| {
```

Comment:
This now returns an error if ledger_db returns an error. Previously, the function would still return JSON but the result would contain null-ish values for any fields that could not be computed.